### ISSUE TITLE
Standing up a flow proxy on google cloud

### ISSUE THREAD

**Issue by Ozzadar** [vfarcic/docker-flow-proxy#18]
Hey,

I'm having issues setting this up on google cloud using the docker-flow-proxy-demo-environments.sh. I've changed the driver to google.

It's provisioning the machines properly but Consul is failing to bind to $DOCKER_IP:8300 and is causing the container to exit on creation. (which causes flow-proxy container to fail)

Any insights on how to get this functioning properly? 

I'll paste the logs in here soon, just have to recreate them.

Thanks!

**Comment by vfarcic** [comment]
Hi,

I am about to board a plane and won't be available for a few days. Send me
the logs and, if they do not show me some obvious problem, I'll set it up
on Google cloud this Friday and get back to you. I hope that is not too
late.

On Tue, May 24, 2016, 15:02 Paul Mauviel <notifications@github.com> wrote:

> Hey,
>
> I'm having issues setting this up on google cloud using the
> docker-flow-proxy-demo-environments.sh. I've changed the driver to google.
>
> It's provisioning the machines properly but Consul is failing to bind to
> $DOCKER_IP:8300 and is causing the container to exit on creation. (which
> causes flow-proxy container to fail)
>
> Any insights on how to get this functioning properly?
>
> I'll paste the logs in here soon, just have to recreate them.
>
> Thanks!
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/vfarcic/docker-flow-proxy/issues/18>
>
-- 
Viktor Farcic

Blog:
TechnologyConversations.com

Books:
The DevOps 2.0 Toolkit (http://www.amazon.com/dp/B01BJ4V66M or
https://leanpub.com/the-devops-2-toolkit)
Test-Driven Java Development (
http://www.amazon.com/Test-Driven-Java-Development-Viktor-Farcic-ebook/dp/B00YSIM3SC
)

**Comment by Ozzadar** [comment]
just got the logs but from the script:

```
To see how to connect your Docker Client to the Docker Engine running on this virtual machine, run: docker-machine env proxy
docker IP: 104.197.207.130
WARNING: The CONSUL_SERVER_IP variable is not set. Defaulting to a blank string.
Creating network "dockerflowproxy_default" with the default driver
Pulling consul-server (consul:latest)...
latest: Pulling from library/consul
d0ca440e8637: Pull complete
a3ed95caeb02: Pull complete
3c4acaac7669: Pull complete
9e29a8caa027: Pull complete
021a3d1c3119: Pull complete
2c0c162d2abb: Pull complete
Digest: sha256:ec46d5123cfe22bd54bf310057d0f862989b977d4122e7dce0d66be647d9f12c
Status: Downloaded newer image for consul:latest
Creating consul
WARNING: The CONSUL_SERVER_IP variable is not set. Defaulting to a blank string.
Pulling proxy (vfarcic/docker-flow-proxy:latest)...
latest: Pulling from vfarcic/docker-flow-proxy
```

From the host:
```
sudo docker logs consul
==> WARNING: BootstrapExpect Mode is specified as 1; this is the same as Bootstrap mode.
==> WARNING: Bootstrap mode enabled! Do not enable unless necessary
==> Starting Consul agent...
==> Error starting agent: Failed to start Consul server: Failed to start RPC layer: listen tcp 104.197.207.130:8300
: bind: cannot assign requested address
```

This is from unmodified .sh file with the exception of the google driver instead of virtualbox.

Hopefully you can help :)

thanks

**Comment by Ozzadar** [comment]
As an addition, I let the script go through to completion and the swarm isnt being connected properly either.

Docker info on swarm-master:3376 is showing no nodes connected so I cant launch anything through swarm.

=(

**Comment by vfarcic** [comment]
A few questions...

I think we could progress faster if we concentrate on Consul only. Can you
please confirm that the ports 8500, 8301, and 8300 are open? Judging by the
logs, seems like they're not. Can you please open those ports and rerun
lines 9 (
https://github.com/vfarcic/docker-flow-proxy/blob/master/docker-flow-proxy-demo-environments.sh#L9),
11 (
https://github.com/vfarcic/docker-flow-proxy/blob/master/docker-flow-proxy-demo-environments.sh#L11),
and 13 (
https://github.com/vfarcic/docker-flow-proxy/blob/master/docker-flow-proxy-demo-environments.sh#L13).
If it turns out that the problem was indeed in ports not being open, later
on we can change the IP from public to private in order to avoid any
security risk.

On Tue, May 24, 2016 at 3:55 PM Paul Mauviel <notifications@github.com>
wrote:

> As an addition, I let the script go through to completion and the swarm
> isnt being connected properly either.
>
> Docker info on swarm-master:3376 is showing no nodes connected so I cant
> launch anything through swarm.
>
> =(
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/vfarcic/docker-flow-proxy/issues/18#issuecomment-221277724>
>
-- 
Viktor Farcic

Blog:
TechnologyConversations.com

Books:
The DevOps 2.0 Toolkit (http://www.amazon.com/dp/B01BJ4V66M or
https://leanpub.com/the-devops-2-toolkit)
Test-Driven Java Development (
http://www.amazon.com/Test-Driven-Java-Development-Viktor-Farcic-ebook/dp/B00YSIM3SC
)

**Comment by Ozzadar** [comment]
I did some messing around and I think I've figured out the problem (though not how to solve it through docker-machine)

So all my ports are open properly. The problem is that it's attempting to bind consul to

$EXTERNAL_IP when the machine it creates doesn't know that this exists. I need to bind to the internal eth0 IP that mirrors to the external ip given to the instance.

Is there any way to resolve the internal IP? I'll probably do a line-by-line manual set up of the whole process to get myself going though being able to set everything up through docker-machine would be ideal.

Thanks

**Comment by Ozzadar** [comment]
Yeah, so I did a little more digging and Im adding something to my script for google.

export CONSUL_INTERNAL=$(gcloud compute instances describe proxy --format=text     | grep '^networkInterfaces\[[0-9]\+\]\.networkIP:' | sed 's/^.* //g')

this will put the internal IP of the proxy docker-machine into $CONSUL_INTERNAL, which I am putting into the Docker-compose.yml instead of $DOCKER_IP in the consul-server definition.

**Comment by Ozzadar** [comment]
Ok, my whole stack is now launching through scripts. 

Is there a known issue where the proxy works in a erratic fashion? I often require several refreshes to get anything except "503 Service Unavailable"

after a few refreshes it loads but it doesn't give me any css.

You can see my whole setup for google here:

https://github.com/Ozzadar/docker-flow-proxy

I'll be setting up the proxy that I had set up previously with consul-template for now seeing as I have enough to get on with for the moment.

**Comment by vfarcic** [comment]
There was a problem in the way the proxy was generating the configuration. This has been fixed in the release 1.0.1. Can you please pull the latest release and let me know whether it works in your setup?
